mcs/class/corlib/System/TimeZoneInfo.cs

```diff
@@ -1347,12 +1361,29 @@ private static TimeZoneInfo ParseTZBuffer (string id, byte [] buffer, int length
 			int typecnt = ReadBigEndianInt32 (buffer, 36);
 			int charcnt = ReadBigEndianInt32 (buffer, 40);
 
+			byte version = buffer[4];
+			int index = 0;
+			int timeValuesLength = 4;
+			if (version == '2' || version == '3') {
```

Comment:
It's available as [a draft](https://www.ietf.org/archive/id/draft-murchison-rfc8536bis-07.html) only, but _technically_ `TZif4` exists (though no sure when `zic` would generate it).

If I read this code correctly, it will fail in the same way as it used to prior to this fix.